tests/test_memory_size.cpp

```diff
@@ -189,8 +230,13 @@ TEST(MemorySize, canCompareEqual)
     EXPECT_EQ(mp::MemorySize{}, mp::MemorySize{"0B"});
     EXPECT_EQ(mp::MemorySize{"2048"}, mp::MemorySize{"2k"});
     EXPECT_EQ(mp::MemorySize{"2g"}, mp::MemorySize{"2048M"});
-    EXPECT_EQ(mp::MemorySize{"2g"}, mp::MemorySize{"2048M"});
     EXPECT_EQ(mp::MemorySize{"0m"}, mp::MemorySize{"0k"});
+    EXPECT_EQ(mp::MemorySize{"1.5G"}, mp::MemorySize{"1536M"});
+    EXPECT_EQ(mp::MemorySize{"1.0K"}, mp::MemorySize{"1024B"});
+    EXPECT_EQ(mp::MemorySize{"1.0K"}, mp::MemorySize{"1k"});
+    EXPECT_EQ(mp::MemorySize{"3.14K"}, mp::MemorySize{"3215"});
+    EXPECT_EQ(mp::MemorySize{"0.0001G"}, mp::MemorySize{"107374"});
+    EXPECT_EQ(mp::MemorySize{"0.095367432K"}, mp::MemorySize{"97B"});
```

Comment:
Nice, you followed the same flooring approach in this case. However, if we accept something like this and discard fractional bytes, why not accept fractional bytes and react the same way? You could then add another test in the sequence of convertsLowerUnitByFlooring (e.g. check that 7.89B, in bytes, is 7).

Alternatively, we could refuse any input that implied fractional bytes, but that would be problematic because we'd then have to refuse something like 3.3K.